variables.tf

```diff
@@ -5,6 +5,16 @@ variable "name_prefix" {
   description = "A prefix used for naming resources."
 }
 
+variable "override_container_name" {
```

Comment:
I think you can omit this variable, and instead _just_ add `container_name` as a variable which has a default value of `""` as it currently does. Then your if statements can just be:

```
container_name   = "${var.container_name != "" ? var.container_name : var.name_prefix}"
```

Reply:
Hi. My initial pull request was actually defined like this, but Colin pointed out that we needed the additional boolean because:
"if var.container_name is itself interpolated before it gets here this line will fail."
So this might still be valid unless there has been changes in terraform that avoids this situation.

Reply:
> "if var.container_name is itself interpolated before it gets here this line will fail."

I think this only applies if you are doing `count` on an interpolated value, otherwise it should be fine @colincoleman? (Also worth noting that this is no longer an issue in terraform `>= 0.12`)